File: src/presentation/pages/foot_selection_page.js
import { footSelectionCard } from '../widgets/foot_selection_card.js';
import { recordingState } from '../state/recording_state.js';

const PROFILE_IMG = 'https://lh3.googleusercontent.com/aida-public/AB6AXuDQ4f0KWiv65lVfJgVKe3zakMCW4-7cr8gCCzRTvhtju1P5CKTNOs0dDoQxxZ5tlWX123e1dRQwGFbU-0y42bLgv3HAmwzB4tVbJh6kc2WR_S8VSljOOOmNnFc05oJbjGJNpYbDQ6HINurTwqyV0ya-BcqxakPwn57kzTirsYOwvQbjtrJ4gTzmxVr-L26h-xDEp1Y9AA37G4NQ3hOWaw3LREPJpHW6Gz_k13rJXHpToUUpbYJ_OO2NYlGLknK6uv96IOnBiMhlrz8';

function buildCards(selectedFoot) {
  return `
    ${footSelectionCard({ foot: 'left',  selected: selectedFoot, description: 'Analyze balance and step pressure for the left side.' })}
    ${footSelectionCard({ foot: 'right', selected: selectedFoot, description: 'Analyze balance and step pressure for the right side.' })}
  `;
}

export function render() {
  return `
    <div class="bg-background text-on-background font-body-md selection:bg-primary-container selection:text-on-primary-container" style="min-height: max(884px, 100dvh);">
      <!-- Header -->
      <header class="bg-surface/80 backdrop-blur-md shadow-sm sticky top-0 z-50 h-16 w-full flex justify-between items-center px-margin-mobile">
        <div class="flex items-center gap-md">
          <button data-nav-to="/instruction" class="transition-all duration-200 active:scale-95 hover:opacity-80 text-primary">
            <span class="material-symbols-outlined">arrow_back</span>
          </button>
          <h1 class="font-headline-sm text-headline-sm text-primary">Gait Analysis</h1>
        </div>
        <div class="w-10 h-10 rounded-full bg-surface-variant flex items-center justify-center overflow-hidden border border-outline-variant/30">
          <img src="${PROFILE_IMG}" alt="User profile photo" class="w-full h-full object-cover">
        </div>
      </header>

      <main class="min-h-[calc(100vh-64px)] px-margin-mobile pt-lg pb-xl max-w-lg mx-auto flex flex-col">
        <!-- Title -->
        <section class="mb-xl text-center">
          <h2 class="font-display-lg text-display-lg text-on-surface mb-xs">Select Foot Side</h2>
          <p class="font-body-md text-body-md text-on-surface-variant">Choose which foot you want to record first.</p>
        </section>

        <!-- Cards -->
        <div id="foot-cards" class="grid grid-cols-1 gap-lg mb-xl">
          ${buildCards(recordingState.selectedFoot)}
        </div>

        <!-- Info Box -->
        <div class="bg-surface-container-high rounded-md p-md mb-xl flex gap-md items-start">
          <span class="material-symbols-outlined text-secondary" style="font-variation-settings: 'FILL' 1">info</span>
          <p class="font-label-md text-label-md text-on-surface-variant">
            You will be asked to record the other foot after this session.
          </p>
        </div>

        <!-- Action Button -->
        <div class="mt-auto">
          <button id="continue-recording-btn" class="w-full bg-primary text-on-primary font-headline-sm py-4 rounded-full shadow-lg transition-all duration-300 active:scale-95 hover:opacity-90 flex items-center justify-center gap-base">
            Continue to Recording
            <span class="material-symbols-outlined">arrow_forward</span>
          </button>
        </div>
      </main>

      <footer class="h-8 w-full bg-transparent"></footer>
    </div>
  `;
}

export function mount(navigate) {
  document.querySelectorAll('[data-nav-to]').forEach(el => {
    el.addEventListener('click', () => navigate(el.dataset.navTo));
  });

  document.querySelectorAll('[data-foot]').forEach(btn => {
    btn.addEventListener('click', () => {
      recordingState.selectedFoot = btn.dataset.foot;
      const container = document.getElementById('foot-cards');
      if (container) container.innerHTML = buildCards(recordingState.selectedFoot);
      // Re-attach after re-render
      mount(navigate);
    });
  });

  document.getElementById('continue-recording-btn')?.addEventListener('click', () => navigate('/recording'));
}

File: lib/presentation/pages/foot_selection_page.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import 'package:provider/provider.dart';
import '../../core/constants/app_colors.dart';
import '../../core/constants/app_routes.dart';
import '../../core/constants/app_text_styles.dart';
import '../../core/theme/app_theme.dart';
import '../providers/recording_provider.dart';
import '../widgets/app_card.dart';

class _BackAppBar extends StatelessWidget {
  final String title;
  const _BackAppBar({required this.title});

  @override
  Widget build(BuildContext context) {
    return Container(
      height: 64 + MediaQuery.of(context).padding.top,
      padding: EdgeInsets.only(
          top: MediaQuery.of(context).padding.top, left: 20, right: 20),
      color: AppColors.surface,
      child: Row(
        children: [
          GestureDetector(
            onTap: () => context.go(AppRoutes.instruction),
            child: Container(
              width: 40,
              height: 40,
              decoration: BoxDecoration(
                color: AppColors.surfaceContainerLow,
                borderRadius: BorderRadius.circular(12),
              ),
              child: const Icon(Icons.arrow_back,
                  size: 20, color: AppColors.onSurface),
            ),
          ),
          const SizedBox(width: 12),
          Text(title, style: AppText.headlineSm),
        ],
      ),
    );
  }
}

class FootSelectionPage extends StatelessWidget {
  const FootSelectionPage({super.key});

  @override
  Widget build(BuildContext context) {
    final provider = context.watch<RecordingProvider>();

    return Scaffold(
      backgroundColor: AppColors.background,
      body: Column(
        children: [
          _BackAppBar(title: 'Select Foot Side'),
          Expanded(
            child: SingleChildScrollView(
              padding: const EdgeInsets.all(AppTheme.marginMobile),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  Text(
                    'Which foot would you like to record first?',
                    style: AppText.bodyMd
                        .copyWith(color: AppColors.onSurfaceVariant),
                  ),
                  const SizedBox(height: AppTheme.lg),
                  // Foot selection cards
                  Row(
                    children: [
                      Expanded(
                        child: _FootCard(
                          label: 'Left Foot',
                          side: 'left',
                          isSelected: provider.selectedFootSide == 'left',
                          onTap: () => provider.selectFootSide('left'),
                        ),
                      ),
                      const SizedBox(width: AppTheme.md),
                      Expanded(
                        child: _FootCard(
                          label: 'Right Foot',
                          side: 'right',
                          isSelected: provider.selectedFootSide == 'right',
                          onTap: () => provider.selectFootSide('right'),
                        ),
                      ),
                    ],
                  ),
                  const SizedBox(height: AppTheme.lg),
                  // Info box
                  Container(
                    padding: const EdgeInsets.all(AppTheme.md),
                    decoration: BoxDecoration(
                      color: AppColors.infoContainer.withAlpha(80),
                      borderRadius: AppTheme.radiusDefault,
                      border: Border.all(
                          color: AppColors.info.withAlpha(60)),
                    ),
                    child: Row(
                      crossAxisAlignment: CrossAxisAlignment.start,
                      children: [
                        const Icon(Icons.info_outline,
                            color: AppColors.info, size: 20),
                        const SizedBox(width: AppTheme.sm),
                        Expanded(
                          child: Text(
                            'You will be asked to record the other foot after completing this one.',
                            style: AppText.bodySm,
                          ),
                        ),
                      ],
                    ),
                  ),
                  const SizedBox(height: 100),
                ],
              ),
            ),
          ),
          // Sticky footer
          Container(
            padding: const EdgeInsets.all(AppTheme.marginMobile),
            decoration: BoxDecoration(
              color: AppColors.surface,
              border: Border(
                top: BorderSide(
                    color: AppColors.outlineVariant.withAlpha(51)),
              ),
            ),
            child: SafeArea(
              top: false,
              child: GestureDetector(
                onTap: () => context.go(AppRoutes.recording),
                child: Container(
                  width: double.infinity,
                  padding: const EdgeInsets.symmetric(vertical: 16),
                  decoration: BoxDecoration(
                    color: AppColors.primary,
                    borderRadius: AppTheme.radiusFull,
                    boxShadow: AppTheme.primaryShadow,
                  ),
                  child: Text(
                    'Continue to Recording',
                    style:
                        AppText.labelMd.copyWith(color: AppColors.onPrimary),
                    textAlign: TextAlign.center,
                  ),
                ),
              ),
            ),
          ),
        ],
      ),
    );
  }
}

class _FootCard extends StatelessWidget {
  final String label;
  final String side;
  final bool isSelected;
  final VoidCallback onTap;

  const _FootCard({
    required this.label,
    required this.side,
    required this.isSelected,
    required this.onTap,
  });

  @override
  Widget build(BuildContext context) {
    return GestureDetector(
      onTap: onTap,
      child: AnimatedContainer(
        duration: const Duration(milliseconds: 200),
        decoration: BoxDecoration(
          color: isSelected
              ? AppColors.primary.withAlpha(13)
              : AppColors.surfaceContainerLowest,
          borderRadius: AppTheme.radiusLg,
          border: Border.all(
            color: isSelected ? AppColors.primary : AppColors.outlineVariant,
            width: isSelected ? 2 : 1,
          ),
          boxShadow: AppTheme.cardShadow,
        ),
        padding: const EdgeInsets.all(AppTheme.lg),
        child: Column(
          children: [
            Container(
              height: 100,
              alignment: Alignment.center,
              child: Column(
                mainAxisAlignment: MainAxisAlignment.center,
                children: [
                  Icon(
                    side == 'left'
                        ? Icons.directions_walk
                        : Icons.directions_walk,
                    size: 80,
                    color: isSelected
                        ? AppColors.primary
                        : AppColors.outlineVariant,
                  ),
                ],
              ),
            ),
            const SizedBox(height: AppTheme.md),
            Text(label,
                style: AppText.labelMd.copyWith(
                    color: isSelected
                        ? const Color.fromARGB(255, 132, 148, 146)
                        : AppColors.onSurface)),
            const SizedBox(height: AppTheme.xs),
            Text(
              isSelected ? 'Selected' : 'Tap to select',
              style: AppText.labelSm.copyWith(
                  color: isSelected
                      ? AppColors.primary
                      : AppColors.onSurfaceVariant),
            ),
            if (isSelected) ...[
              const SizedBox(height: AppTheme.sm),
              Container(
                width: 28,
                height: 28,
                decoration: const BoxDecoration(
                  color: AppColors.primary,
                  shape: BoxShape.circle,
                ),
                child: const Icon(Icons.check,
                    size: 16, color: AppColors.onPrimary),
              ),
            ],
          ],
        ),
      ),
    );
  }
}
